File: client/frontend/src/types/scheduler.ts
import { Task } from './task'

export enum SchedulerType {
    FCFS = 'FCFS',
    SJF = 'SJF',
    RR = 'RR'
}

export interface ScheduledTask {
    id: string
    name: string
    startTime: Date
    endTime: Date
    estimatedTime: number
    priority: number
}

export interface ScheduleEvaluation {
    totalTime: number
    averageWaitingTime: number
    averageTurnaroundTime: number
    efficiency: number
}

export interface ScheduleResult {
    scheduledTasks: ScheduledTask[]
    evaluation: ScheduleEvaluation
}

export interface SchedulerConfig {
    type: SchedulerType
    timeSlice?: number  // 用于RR算法的时间片大小
}

export interface Scheduler {
    schedule(tasks: Task[]): ScheduleResult[];
    evaluate(tasks: Task[]): ScheduleEvaluation;
} 

File: client/backend/src/schemas/scheduler.py
from pydantic import BaseModel
from typing import List
from datetime import datetime

class ScheduleRequest(BaseModel):
    algorithm: str
    tasks: List[dict]

class ScheduledTask(BaseModel):
    id: str
    name: str
    start_time: datetime
    end_time: datetime
    estimated_time: int
    priority: int

class ScheduleEvaluation(BaseModel):
    total_time: float
    average_waiting_time: float
    average_turnaround_time: float
    efficiency: float

class ScheduleResult(BaseModel):
    scheduled_tasks: List[ScheduledTask]
    evaluation: ScheduleEvaluation 